lib/global_id/global_id.rb

```diff
@@ -43,19 +46,22 @@ def parse_encoded_gid(gid, options)
 
   def initialize(gid, options = {})
     @uri = gid.is_a?(URI::GID) ? gid : URI::GID.parse(gid)
+    validate_model_id
```

Comment:
Why we need this validation? What are we trying to protect?

Reply:
We are trying to prevent someone manually crafting a GID like `gid://app/Person/1/2/3/4/5....` which will be parsed as `[1,2,3,4,5]` and passed to `find()` which may lead to more than one record being loaded.

As an alternative we can move the check to the locator but then custom locators won't be protected. Which may not be a bad thing

Let me know if you'd like to completely loosen the restriction or locator would be a better place to ensure we only target single object when calling `find`

Reply:
GlobalID is primarily used as a serialization format. Someone will not manually be creating GIDs. If people are using GlobalID not a serialization format, then they will be responsible for checking if their input is correct.

I think this is better handled at the locator level.

Reply:
Done! Moved the validation to the locator level. Instead of raising it now returns `nil` as this seems to be locator's interface when it comes to locating records by a malformed gid